Refactor Bigbang's hashing on arm64 platform

ouyun opened the pull request:
Refactor Bigbang's hashing on arm64 platform.

ouyun committed:
Update build arguments for aarch64 with cryto feature to operate hash

ouyun committed:
Add build parameter to fix bug of aes error

ouyun committed:
debug

ouyun committed:
debug

ouyun committed:
debug

ouyun committed:
Refactor cn_slow_hash() for aarch64 with crypto feature according to x86 algo

ouyun committed:
update from IDE

ouyun committed:
Update

ouyun committed:
Refactoring

ouyun committed:
Revise

ouyun committed:
Update code of passing through hash using fallback aes version

ouyun committed:
Update

ouyun committed:
Update

ouyun committed:
Update

ouyun committed:
Update

ouyun committed:
Add unit test for slow hash

ouyun committed:
Merge branch 'master' into powarm

ouyun committed:
Fixup ut

ouyun committed:
Add time measurement

ouyun committed:
Update measuring time for fallback instruction

ouyun committed:
Substitude neon instruction for fallback one

ouyun committed:
Isolate intel sse2 native instructions from arm64 platform

ouyun committed:
bugfix of lacking left brace

ouyun committed:
Update UT

ouyun committed:
Update for resolving phase 2

ouyun committed:
Sync update with x86 version for phase 2

ouyun committed:
Refactor phase 2 hashing

ouyun committed:
Clean up and format

ouyun committed:
Workaround solution for crashing on release version

ouyun committed:
Refactor

ouyun committed:
Remove unused

ouyun committed:
Fixup crash when running release version

ouyun committed:
Update UT

ouyun committed:
Add new height to UT

ouyun committed:
Update build script and refactor it according to ShellCheck

ouyun committed:
Merge branch 'master' into powarm

ouyun committed:
Merge branch 'master' into powarm

ouyun committed:
Remove verificaiton for non-POW blocks in UT

ouyun committed:
Remove unused flag and fix typos

AlexiaChen commented on CMakeLists.txt:
```diff
@@ -56,6 +56,7 @@ endif()
 if(UNIX)
     string(APPEND CMAKE_CXX_FLAGS_DEBUG " -O0")
     string(APPEND CMAKE_CXX_FLAGS_RELEASE " -O2 -DNDEBUG")
+    # string(APPEND CMAKE_CXX_FLAGS_RELEASE " -O0")
```
这里直接删除掉，就不用注释了。

AlexiaChen commented on src/bigbang/core.cpp:
```diff
@@ -834,10 +835,45 @@ Errno CCoreProtocol::VerifyProofOfWork(const CBlock& block, const CBlockIndex* p
     {
         return DEBUG(ERR_BLOCK_PROOF_OF_WORK_INVALID, "hash error: proof[%s] vs. target[%s] with bits[%d]",
                      hash.ToString().c_str(), hashTarget.ToString().c_str(), nBits);
+//        DEBUG(ERR_BLOCK_PROOF_OF_WORK_INVALID, "monero hash error: proof[%s] vs. target[%s] with bits[%d]",
```
这里同上

AlexiaChen commented on test/slowhash_tests.cpp:
```diff
@@ -0,0 +1,81 @@
+#include "uint256.h"
+#include "util.h"
+#include "crypto.h"
+//#include "pow_hash/hash-ops.h"
```
这里同上

AlexiaChen commented on test/slowhash_tests.cpp:
```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019-2020 The Bigbang developers
```
这里是2019-2021，到时候你合并master就可以了

ouyun committed:
Update according by comments from reviewer

ouyun committed:
Update in terms of reviewer:

1, Clean up code;
2, Revert changes of install script for splitting it to another new PR;
